exonum-java-proofs/src/main/java/com/exonum/binding/storage/proofs/map/flat/CheckedMapProofEntry.java

```diff
@@ -0,0 +1,42 @@
+package com.exonum.binding.storage.proofs.map.flat;
+
+import java.util.Arrays;
+
+/**
+ * A map entry corresponding to a leaf node in the map tree.
```

Comment:
I'd use this because this class does not have any connection to a leaf node:
```
A map entry: a key-value pair. This entry does not permit null keys and values
```